object/object.go

```diff
@@ -72,8 +72,9 @@ func GenerateEncryptionKey() EncryptionKey {
 
 // An Object is a unit of data that has been stored on a host.
 type Object struct {
-	Key   EncryptionKey `json:"key"`
-	Slabs []SlabSlice   `json:"slabs"`
+	Key         EncryptionKey `json:"key"`
+	Slabs       []SlabSlice   `json:"slabs"`
+	PartialSlab *PartialSlab  `json:"partial_slab,omitempty"`
```

Comment:
This is the only thing I'm not 100% sure about yet. We essentially use JSON for potentially large amounts of binary data which is not ideal. At the same time this is quite nice because we don't need more endpoints or roundtrips during uploads and downloads.